ams/server/src/main/java/com/netease/arctic/server/dashboard/ServerTableDescriptor.java

```diff
@@ -0,0 +1,313 @@
+          f.content(),
+          f.fileSizeInBytes(),
+          snapshotTime,
+          "add"));
+    });
+    snapshot.removedDeleteFiles(arcticTable.io()).forEach(f -> {
+      result.add(new AMSDataFileInfo(
+          f.path().toString(),
+          arcticTable.spec(), f.partition(),
+          f.content(),
+          f.fileSizeInBytes(),
+          snapshotTime,
+          "remove"));
+    });
+    return result;
+  }
+
+  public List<DDLInfo> getTableOperations(ServerTableIdentifier tableIdentifier) {
+    List<DDLInfo> result = new ArrayList<>();
+    ArcticTable arcticTable = tableService.loadTable(tableIdentifier);
+    Table table;
+    if (arcticTable.isKeyedTable()) {
+      table = arcticTable.asKeyedTable().baseTable();
+    } else {
+      table = arcticTable.asUnkeyedTable();
+    }
+    List<HistoryEntry> snapshotLog = ((HasTableOperations) table).operations().current().snapshotLog();
+    List<org.apache.iceberg.TableMetadata.MetadataLogEntry> metadataLogEntries =
+        ((HasTableOperations) table).operations().current().previousFiles();
+    Set<Long> time = new HashSet<>();
+    snapshotLog.forEach(e -> time.add(e.timestampMillis()));
+    for (int i = 1; i < metadataLogEntries.size(); i++) {
+      org.apache.iceberg.TableMetadata.MetadataLogEntry e = metadataLogEntries.get(i);
+      if (!time.contains(e.timestampMillis())) {
+        org.apache.iceberg.TableMetadata
+            oldTableMetadata = TableMetadataParser.read(table.io(), metadataLogEntries.get(i - 1).file());
+        org.apache.iceberg.TableMetadata
+            newTableMetadata = TableMetadataParser.read(table.io(), metadataLogEntries.get(i).file());
+        DDLInfo.Generator generator = new DDLInfo.Generator();
+        result.addAll(generator.tableIdentify(arcticTable.id())
+            .oldMeta(oldTableMetadata)
+            .newMeta(newTableMetadata)
+            .generate());
+      }
+    }
+    if (metadataLogEntries.size() > 0) {
+      org.apache.iceberg.TableMetadata oldTableMetadata = TableMetadataParser.read(
+          table.io(),
+          metadataLogEntries.get(metadataLogEntries.size() - 1).file());
+      org.apache.iceberg.TableMetadata newTableMetadata = ((HasTableOperations) table).operations().current();
+      DDLInfo.Generator generator = new DDLInfo.Generator();
+      result.addAll(generator.tableIdentify(arcticTable.id())
+          .oldMeta(oldTableMetadata)
+          .newMeta(newTableMetadata)
+          .generate());
+    }
+    return result;
+  }
+
+  public List<BaseMajorCompactRecord> getOptimizeInfo(String catalog, String db, String table) {
+    List<TableOptimizingProcess> tableOptimizingProcesses = getAs(
+        OptimizingMapper.class,
+        mapper -> mapper.selectSuccessOptimizingProcesses(catalog, db, table));
+    return tableOptimizingProcesses.stream().map(e -> {
+      BaseMajorCompactRecord record = new BaseMajorCompactRecord();
+      List<TaskRuntime> taskRuntimes = getAs(
```

Comment:
Is it better to use join sql? As the number of process records increases, the execution time will be enlarged.